Updates the DL definition

LJWatson opened the pull request:
* Improves readability (hopefully)
* Removes inaccurate and/or incorrect information
* Adds explanation and examples for div elements
* Fixes #977 #1116

LJWatson committed:
Updates the DL definition
* Improves readability (hopefully)
* Removes inaccurate and/or incorrect information
* Adds explanation and examples for div elements
* Fixes #977 #1116

chaals commented on sections/semantics-grouping-content.include:
```diff
@@ -1226,12 +1226,16 @@
 
-  The <{dl}> element <a>represents</a> a description list of zero or more <dfn>term-description groups</dfn>.
+    The <{dl}> element <a>represents</a> a description list of zero or more <dfn>term-description
+  groups</dfn>.
```
Please keep the `dfn` element on a single line, so people can search the source for it.

chaals commented on sections/semantics-grouping-content.include:
```diff
@@ -1248,10 +1252,13 @@
   </div>
 
-  Each term within a <a>term-description group</a> must be represented by a single <{dt}> element. The descriptions within a <a>term-description group</a> are alternatives. Each description must be represented by a single <{dd}> element.
+  Each term within a <a>term-description group</a> must be represented by a single <{dt}> element.
```
Which means that we have an HTML validity error in many algorithm switches, because they use multiple `dt`s as alternatives to be addressed by a single following `dd`.

I'll raise a separate issue on that. EIther we are doing the wrong thing, or the spec is overly prescriptive and unhelpful, but I don't know yet...

chaals replied:
Hmm. Reading again, I see that it does allow multiple terms in the same group.

chaals reviewed (changes requested):
one whitespace tweak - I'll commit that, an issue to raise as a consequence of this, and a request for more work...

Please explain what "ignored" means in the context of this section: will not be parsed, will not be rendered, will be rendered outside the `dl` in the way stray elements in a `table` are, or something else.

chaals committed:
keep term name on one line

It helps with searching through source. Which I do a lot, and find that changes get missed when that fails.

LJWatson committed:
Removes confusing language from DL definition

LJWatson commented:
<Which means that we have an HTML validity error in many algorithm switches, because they use multiple dts as alternatives to be addressed by a single following
dd.<

I don't think there is anything in the definition that prevents multiple ```<dt>``` elements. It says `each` term must be represented by a single ```<dt>``` element, not that there must be only one ```<dt>``` element.

scottaohara committed:
changes &amp;c to &c

minor update concerning the previous work of changing `pre` elements to `xmp` for examples.

scottaohara committed:
updates to descriptions of divs within dls

Worked with Léonie to revise some wording for divs within dls.

Also fix double instance of "note:"

scottaohara reviewed (approved):
worked with Léonie to make some prose revisions.  Pending other requested edits, this looks good to me.
